parsl/executors/radical/executor.py

```diff
@@ -186,6 +194,23 @@ def task_state_cb(self, task, state):
                     else:
                         parsl_task.set_exception('Task failed for an unknown reason')
 
+    def _fail_all_tasks(self, exception):
+        """
+        Fail all outstanding tasks with the given exception.
+
+        This method iterates through all outstanding tasks in the
+        `_future_tasks` dictionary, which have not yet completed,
+        and sets the provided exception as their result, indicating
+        a failure.
+
+        Parameters:
+        - exception: The exception to be set as the result for all
+                     outstanding tasks.
+        """
+        for fut_task in self.future_tasks.values():
+            if not fut_task.done():
+                fut_task.set_exception(exception)
```

Comment:
maybe there's a very slight race condition here - can fut_task get set by another thread (eg. task completion) in the space between line 211 and line 212? If so, set_exception would raise an exception here. I've seen a few situations like that elsewhere in Parsl.